packages/plugin-cache-memory/src/memory.ts

```diff
@@ -133,7 +151,13 @@ export default ({
             const cacheKey = getCacheKeyUtil(context, getCacheKey);
             const ttl: number = prop('memoryCacheTtl', context.getRequest());
 
-            lruCache.set(cacheKey, context.getResponse(), { ttl });
+            const httpMeta = context.getInternalMeta(metaTypes.PROTOCOL_HTTP);
+
+            const value = {
+                response: context.getResponse(),
+                status: httpMeta?.response?.status
```

Comment:
I think we shouldn't fill the cache in case of `undefined` status. It's not about type safety only, but looks like the main purpose of this PR – is to make guarantees, that cached status never be `undefined`.

Or we must consider it in `StoredValue` interface.

Reply:
Sounds reasonable, but I don't want to bind this plugin to the http protocol only, so the second option seems like a solution to me